From a765377fc3a698fdf40d486820c7f5149d660b3f Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Fri, 26 Jul 2019 18:55:44 +0800
Subject: [PATCH] Add WRLinux1019 specific code for rule
 require_singleuser_auth

Add WRLinux1019 relevant specific code for rule require_singleuser_auth.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../require_singleuser_auth/bash/wrlinux1019.sh    | 11 +++
 .../require_singleuser_auth/oval/wrlinux1019.xml   | 91 ++++++++++++++++++++++
 2 files changed, 102 insertions(+)
 create mode 100644 linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/bash/wrlinux1019.sh
new file mode 100644
index 0000000..2dbb645
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/bash/wrlinux1019.sh
@@ -0,0 +1,11 @@
+# platform = multi_platform_wrlinux
+
+service_file="/lib/systemd/system/rescue.service"
+
+sulogin="/usr/lib/systemd/systemd-sulogin-shell"
+
+if grep "^ExecStart=.*" "$service_file" ; then
+    sed -i "s%^ExecStart=.*%ExecStart=-$sulogin rescue%" "$service_file"
+else
+    echo "ExecStart=-$sulogin rescue" >> "$service_file"
+fi
diff --git a/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/oval/wrlinux1019.xml b/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/oval/wrlinux1019.xml
new file mode 100644
index 0000000..717fb64
--- /dev/null
+++ b/linux_os/guide/system/accounts/accounts-physical/require_singleuser_auth/oval/wrlinux1019.xml
@@ -0,0 +1,91 @@
+<def-group>
+  <definition class="compliance" id="require_singleuser_auth" version="1">
+    <metadata>
+      <title>Require Authentication for Single-User Mode</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The requirement for a password to boot into single-user mode
+      should be configured correctly.</description>
+    </metadata>
+    {{%- if init_system == "systemd" -%}}
+    <criteria operator="AND">
+      <criterion comment="Conditions are satisfied"
+      test_ref="test_require_rescue_service" />
+      <criterion test_ref="test_require_rescue_service_runlevel1" />
+      <criterion test_ref="test_no_custom_runlevel1_target" negate="true"/>
+      <criterion test_ref="test_no_custom_rescue_service" negate="true"/>
+    </criteria>
+    {{%- else -%}}
+    <criteria>
+      <criterion comment="Conditions are satisfied"
+      test_ref="test_require_singleuser_auth" />
+    </criteria>
+    {{%- endif -%}}
+  </definition>
+  {{%- if init_system == "systemd" -%}}
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+    comment="Tests that
+    {{% if product in ["fedora", "rhel8"] -%}}
+    /usr/lib/systemd/systemd-sulogin-shell
+    {{%- else -%}}
+    /sbin/sulogin
+    {{%- endif %}}
+    was not removed from the default systemd rescue.service to ensure that a
+  password must be entered to access single user mode"
+  id="test_require_rescue_service" version="1">
+    <ind:object object_ref="obj_require_rescue_service" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_require_rescue_service" version="1">
+    <ind:filepath>/lib/systemd/system/rescue.service</ind:filepath>
+    <ind:pattern operation="pattern match">^ExecStart=\-.*/usr/lib/systemd/systemd-sulogin-shell[ ]+rescue</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+    comment="Tests that the systemd rescue.service is in the runlevel1.target"
+    id="test_require_rescue_service_runlevel1" version="1">
+    <ind:object object_ref="obj_require_rescue_service_runlevel1" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_require_rescue_service_runlevel1" version="1">
+    <ind:filepath>/lib/systemd/system/runlevel1.target</ind:filepath>
+    <ind:pattern operation="pattern match">^Requires=.*rescue.service</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <unix:file_test check="all" check_existence="at_least_one_exists" 
+  comment="look for rescue.service in /etc/systemd/system" 
+  id="test_no_custom_rescue_service" version="1">
+    <unix:object object_ref="object_no_custom_rescue_service" />
+  </unix:file_test>
+  <unix:file_object comment="look for rescue.service in /etc/systemd/system"
+  id="object_no_custom_rescue_service" version="1">
+    <unix:behaviors recurse="directories" recurse_direction="down" recurse_file_system="all" />
+    <unix:path operation="equals">/etc/systemd/system</unix:path>
+    <unix:filename operation="pattern match">^rescue.service$</unix:filename>
+  </unix:file_object>
+
+  <unix:file_test check="all" check_existence="at_least_one_exists" 
+  comment="look for runlevel1.target in /etc/systemd/system"
+  id="test_no_custom_runlevel1_target" version="1">
+    <unix:object object_ref="object_no_custom_runlevel1_target" />
+  </unix:file_test>
+  <unix:file_object comment="look for runlevel1.target in /etc/systemd/system"
+  id="object_no_custom_runlevel1_target" version="1">
+    <unix:behaviors recurse="directories" recurse_direction="down" recurse_file_system="all" />
+    <unix:path operation="equals">/etc/systemd/system</unix:path>
+    <unix:filename operation="pattern match">^runlevel1.target$</unix:filename>
+  </unix:file_object>
+  {{%- else -%}}
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="Tests that the SINGLE variable in the /etc/sysconfig/init file is set to /sbin/sulogin, to ensure that a password must be entered to access single user mode"
+  id="test_require_singleuser_auth" version="1">
+    <ind:object object_ref="obj_require_singleuser_auth" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_require_singleuser_auth" version="1">
+    <ind:filepath>/etc/sysconfig/init</ind:filepath>
+    <ind:pattern operation="pattern match">^SINGLE=/sbin/sulogin[\s]*</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+  {{%- endif -%}}
+</def-group>
-- 
1.9.1

